data-model/src/main/java/io/micronaut/data/model/query/builder/AbstractSqlLikeQueryBuilder.java

```
@@ -1643,6 +1643,13 @@ public int getParameterIndex() {
                 QueryPropertyPath propertyPath = entry.getKey();
                 if (entry.getValue() instanceof BindingParameter bindingParameter) {
                     traversePersistentProperties(propertyPath.getAssociations(), propertyPath.getProperty(), (associations, property) -> {
+                        boolean generated = property.isGenerated();
```

This check was not present for update, but if we want to skip updating generated values then this is needed. Will skip if property being traversed is generated and belongs to this root entity or is inside embedded entity.

---

Btw, in original issue reproducer example since all non id fields are `@GeneratedValue` then update would produce `UPDATE SET WHERE id = :id` and will fail so we might want to throw an exception telling there are no fields to update or something like that? Can't make up reasonable error message for this case.

---

That would be major change and the error would be anyway obvious to the user why update failed in such case so leaving this for now